src/test/java/com/hillel/items_exchange/service/UserServiceTest.java

```diff
@@ -0,0 +1,166 @@
+    }
+
+    @Test
+    void testUpdateUserPassword_WhenPasswordConfirmationWrong_ShouldThrowIllegalArgumentException() {
+        //given
+        UserChangePasswordDto userChangePasswordDto =
+                createUserChangePasswordDtoWithWrongPasswordConfirmation();
+        //then
+        assertThrows(IllegalArgumentException.class,
+                () -> userService.updateUserPassword(userChangePasswordDto, userWithOldPassword));
+    }
+
+    @Test
+    void testUpdateUserEmail_Successfully() {
+        //given
+        UserChangeEmailDto userChangeEmailDto =
+                createUserChangeEmailDtoWithCorrectData();
+        User userWithNewEmail = createUserWithNewEmail();
+        //when
+        when(userRepository.saveAndFlush(userWithOldPassword)).thenReturn(userWithNewEmail);
+        UserDto userDto = userService.updateUserEmail(userChangeEmailDto, userWithOldPassword);
+        //then
+        assertEquals(userDto.getUsername(), userWithNewEmail.getUsername());
+        verify(userRepository).saveAndFlush(userWithOldPassword);
+    }
+
+    @Test
+    void testUpdateUserEmail_WhenEmailConfirmationWrong_ShouldThrowIllegalArgumentException() {
+        //given
+        UserChangeEmailDto userChangeEmailDto =
+                createUserChangeEmailDtoWithWrongEmailConfirmation();
+        //then
+        assertThrows(IllegalArgumentException.class,
+                () -> userService.updateUserEmail(userChangeEmailDto, userWithOldPassword));
+    }
+
+    private User createUserWithOldPassword() {
+        userWithOldPassword = new User();
+        userWithOldPassword.setPassword(bCryptPasswordEncoder.encode(CORRECT_OLD_PASSWORD));
+
+        return userWithOldPassword;
+    }
+
+    private User createUserWithNewPassword() {
+        User user = new User();
+        user.setPassword(NEW_PASSWORD);
+
+        return user;
+    }
+
+    private UserChangePasswordDto createUserChangePasswordDtoWithCorrectData() {
+
+        return createUserChangePasswordDto(CORRECT_OLD_PASSWORD,
+                NEW_PASSWORD,
+                NEW_PASSWORD);
+    }
+
+    private UserChangePasswordDto createUserChangePasswordDtoWithWrongOldPassword() {
+
+        return createUserChangePasswordDto(WRONG_OLD_PASSWORD,
+                NEW_PASSWORD,
+                NEW_PASSWORD);
+    }
+
+    private UserChangePasswordDto createUserChangePasswordDtoWithWrongPasswordConfirmation() {
+
+        return createUserChangePasswordDto(CORRECT_OLD_PASSWORD,
+                NEW_PASSWORD,
+                WRONG_NEW_PASSWORD_CONFIRMATION);
+    }
+
+    private UserChangePasswordDto createUserChangePasswordDto(String password,
+                                                              String newPassword,
+                                                              String confirmNewPassword) {
+        return new UserChangePasswordDto(password, newPassword, confirmNewPassword);
+    }
+
+    private User createUserWithNewEmail() {
```

Comment:
Just create a builder and it will allow you to invoke it with parameters that you really need.